management_api.go

```diff
@@ -68,6 +92,32 @@ func handleStatus(responseWriter http.ResponseWriter, request *http.Request, ser
 			ResourceRequirements: service.ResourceRequirements,
 		}
 
+		// Determine service URL template to use
+		var urlTemplate string
+		if service.ServiceUrl != nil && service.ServiceUrl.IsSet() {
+			if service.ServiceUrl.IsNull() {
+				// ServiceUrl explicitly set to null - no URL desired
+				urlTemplate = ""
+			} else {
+				// ServiceUrl set to a specific template
+				urlTemplate = service.ServiceUrl.Value()
+			}
+		} else if config.DefaultServiceUrl != nil {
+			// ServiceUrl not specified - fall back to DefaultServiceUrl
+			urlTemplate = *config.DefaultServiceUrl
+		}
+
+		// Render service URL if template is available and not empty
+		// Only skip rendering if ServiceUrl was explicitly set to null
+		if urlTemplate != "" && service.ListenPort != "" && !(service.ServiceUrl != nil && service.ServiceUrl.IsSet() && service.ServiceUrl.IsNull()) {
+			renderedUrl, err := renderServiceUrl(urlTemplate, service.ListenPort)
+			if err != nil {
+				log.Printf("Failed to render service URL template for service %s: %v", service.Name, err)
```

Comment:
```suggestion
				log.Printf("[Management API] Failed to render service URL template for service %s: %v", service.Name, err)
```

I am trying to have all the logs that are specific to a service to have the service name in the square brackets.

Reply:
Doing this in my fixup PR, but we should do this properly with structured logs: https://go.dev/blog/slog